__tests__/commands/run-integration.js

```diff
@@ -0,0 +1,57 @@
+/* @flow */
+
+import makeTemp from '../_temp.js';
+import * as fs from '../../src/util/fs.js';
+import {run as buildRun} from './_helpers.js';
+import {BufferReporter} from '../../src/reporters/index.js';
+import {run} from '../../src/cli/commands/run.js';
+
+const path = require('path');
+
+const fixturesLoc = path.join(__dirname, '..', 'fixtures', 'run');
+const runRun = buildRun.bind(null, BufferReporter, fixturesLoc, (args, flags, config, reporter): Promise<void> => {
+  return run(config, reporter, flags, args);
+});
+
+jasmine.DEFAULT_TIMEOUT_INTERVAL = 30000;
```

Comment:
Well, now I see. Actually, I think you can benefit from using `test.concurrent` and splitting your tests into individual test cases, generated programmatically, wrap them in a `describe` block so you can use `beforeEach` hooks and then `jest` should take care of it. If even after that and not writing to the file system (remember, we agreed to try using stdout instead above?) I'm okay leaving the timeout higher.